pkp/pkp-lib#2089 orphaned submissions in the active grid

bozana opened the pull request:
s. https://github.com/pkp/pkp-lib/issues/2089

asmecher commented on controllers/grid/submissions/activeSubmissions/ActiveSubmissionsListGridHandler.inc.php:
```diff
@@ -100,7 +100,7 @@ protected function loadData($request, $filter) {
 
 		$nonExistingUserId = 0;
-		return $submissionDao->getActiveSubmissions($context->getId(), $title, $author, $editor, $stageId, $rangeInfo);
+		return $submissionDao->getActiveSubmissions($context->getId(), $title, $author, $editor, $stageId, $rangeInfo, true);
```
This additional condition reminds me of the "show users with no roles in this journal" checkbox in the users area. I wonder if it's worth adding a filter checkbox for this condition?

bozana replied:
Yes, maybe, if it can be checked per default -- I think it would be good to immediately see them...

asmecher commented on classes/submission/SubmissionDAO.inc.php:
```diff
@@ -820,7 +821,8 @@ function getActiveSubmissions($contextId = null, $title = null, $author = null,
 				' . ($stageId?' AND s.stage_id = ?':'') . '
-				' . ($editor?' AND (g.role_id = ? OR g.role_id = ?) AND' . $this->_getEditorSearchQuery():'') .
+				' . ($editor?' AND (g.role_id = ? OR g.role_id = ?) AND' . $this->_getEditorSearchQuery():'') . '
+				' . ($orphaned?' OR (s.submission_progress = 1 AND s.submission_id NOT IN (SELECT sa2.submission_id FROM stage_assignments sa2 WHERE sa2.user_group_id = ' . (int) ROLE_ID_AUTHOR .'))':'') .
```
Why `submission_progress = 1` and not `submission_progress <> 0`?

bozana replied:
Ah, this should go above, where s.date_submitted in not null or ...

bozana commented:
@asmecher, I corrected the SQL and added a new filter for active submissions grid. I didn't manage to have that filter option checked per default, but it's OK so, or? Thanks and sorry for the first quick and wrong SQL!

asmecher commented:
That's what code reviews are for! This looks good -- just one stylistic comment, feel free to take it or leave it, and go ahead with merging when you're ready.

bozana commented:
Your wish, my command :-)
